config/docker.spc

```diff
@@ -1,12 +1,23 @@
 connection "docker" {
   plugin = "docker"
 
-  # Paths is a list of locations to search for Dockerfiles by default.
-  # Wildcards are supported per https://golang.org/pkg/path/filepath/#Match
-  # Exact file paths can have any name. Wildcard based matches must either
-  # have a name of Dockerfile (e.g. Dockerfile, Dockerfile.example) or an
-  # .dockerfile extension (e.g. nginx.dockerfile).
-  # paths = [ "/path/to/dir/*", "/path/to/exact/custom-dockerfile-name" ]
+  # Paths is a list of locations to search for Dockerfiles
+  # All paths are resolved relative to the current working directory (CWD)
+  # Wildcard based searches are supported, including recursive searches
+
+  # For example:
+  #  - "*.dockerfile" matches all Dockerfiles in the CWD
+  #  - "**/*.dockerfile" matches all Dockerfiles in the CWD and all sub-directories
+  #  - "../*.dockerfile" matches all Dockerfiles in the CWD's parent directory
+  #  - "Dockerfile" matches all Dockerfiles named "Dockerfile" in the CWD
+  #  - "/path/to/dir/*.dockerfile" matches all Dockerfiles in a specific directory
+  #  - "/path/to/dir/Dockerfile" matches a specific Dockerfile
+
+  # If paths includes "*", all files (including non-Dockerfiles) in
+  # the CWD will be matched, which may cause errors if incompatible file types exist
+
+  # Defaults to CWD
+  paths = [ "Dockerfile", "*.dockerfile" ]
```

Comment:
Should the first default here be `Dockerfile*` or `Dockerfile`? What's common naming practice for Dockerfiles?

Also, is matching case sensitive for instances of `Dockerfile` and `*.dockerfile`, e.g., will dockerfile.test and test.Dockerfile match respectively?

Reply:
Based on current defaults when loading files, we'll use `Dockerfile` and `*.dockerfile` as the defaults and see if any users have feedback. The `paths` argument can be modified as well per connection.